### ISSUE TITLE
Failed to run vcpkg after bootstrap

### ISSUE THREAD

**Issue by roalz** [Microsoft/vcpkg#866]
After a few months trying to use NuGet for native C++ dependencies, I finally decided to try out vcpkg as a replacement for nuget.  
So I opened a Widnows cmd console (run as Administrator, but I first tried running as normal user),  
checked the prerequisites (I have both VS2015 Update 3 and VS2017 installed, cmake and git for Windows),  
then I followed the steps described in the README.md file:

```
git clone https://github.com/Microsoft/vcpkg.git
cd vcpkg
powershell -exec bypass scripts\bootstrap.ps1
.\vcpkg integrate install
```

No matter what command line option I specify, running vcpkg always produce the following error output (in red):

> no such file or directory

I'm stuck with vcpkg, any idea about what the problem can be, or anything I can try?

**Comment by smasherprog** [comment]
I am receiving the same error, I had to revert back a couple of commits

**Comment by albertziegenhagel** [comment]
The regression has just been introduced a couple of hours again. Take a look at #868 for a resolution of the problem.

**Comment by ras0219-msft** [comment]
@albertziegenhagel Thank you very much for the detailed analysis in the other issue! This made it much easier to fix quickly 👍.

I've reworked that section of code to check the expected value using the normal `if (auto x = p.get()) { use x; }` pattern elsewhere in the code. I've also audited every other try/catch block in the code to ensure that none of them were impacted by the replacement of get_or_throw -> get_or_exit.
